pkgs/development/haskell-modules/generic-builder.nix

```diff
@@ -214,6 +214,10 @@ stdenv.mkDerivation ({
 
   LANG = "en_US.UTF-8";         # GHC needs the locale configured during the Haddock phase.
 
+  # XXX: workaround for https://ghc.haskell.org/trac/ghc/ticket/11042.
+  LD_LIBRARY_PATH = stdenv.lib.makeLibraryPath (stdenv.lib.filter (x: x != null) systemBuildInputs);
```

Comment:
I think it's a bad idea to set `LD_LIBRARY_PATHS` in the generic builder. The `haskell-gi` build is supposed to fail, honestly, because upstream calls their doctest suite with incorrect flags. I don't want us to add kludges to our code so that their broken code can succeed. The proper solution is to fix the `haskell-gi` build. I've committed 3e05301c94d72979da3c5f304196c7dd01f76a15 to accomplish that until a new release comes out.